dvc_webhdfs/__init__.py

```diff
@@ -49,6 +49,18 @@ def _prepare_credentials(self, **config):
     def fs(self):
         from fsspec.implementations.webhdfs import WebHDFS
 
+        # If target data_proxy provided construct the source from host/port
+        if "data_proxy_target" in self.fs_args:
+            host = self.fs_args["host"]
+            port = self.fs_args["port"]
+
+            protocol = "https" if self.fs_args.get("use_https") else "http"
+
+            source_url = f"{protocol}://{host}:{port}/webhdfs/v1"
+            self.fs_args["data_proxy"] = {
+                source_url: self.fs_args["data_proxy_target"]
+            }
+

```

Comment:
Seems like we need to do this in `_prepare_credentials` above instead.

Reply:
I saw that some things happened in the `_prepare_credentials`, but since this had nothing to do with credentials I did not include it in there. Happy to move it though

Reply:
The name is unfortunate, I agree. We use it for transforming our config to fs config. E.g. https://github.com/iterative/dvc-s3/blob/d153738232ac1a6b3920ee7a72b9004f07c193d2/dvc_s3/__init__.py#L131 So let's move it there indeed, please.

Reply:
let me know if you want me to move it and I will add a commit for it

Reply:
I just pushed a new version. It looks ok to me (famous last words :wink:) but will need to wait till tomorrow till I can give this modified code a test at work

Reply:
Sounds good. Then waiting for confirmation from your side.

Reply:
Did a test with this new code and was able to push and pull data from the remote WebHDFS server